nifi-nar-bundles/nifi-h2o-record-bundle/nifi-h2o-record-processors/src/main/java/org/apache/nifi/processors/h2o/record/ExecuteMojoScoringRecord.java

```diff
@@ -0,0 +1,383 @@
+				
+				final Record scoredFirstRecord = predict(firstRecord, model, getLogger());
+				
+				if(scoredFirstRecord == null) {
+					throw new ProcessException("Error scoring the first record");
+				}
+				
+				final RecordSchema writeSchema = writerFactory.getSchema(original.getAttributes(), scoredFirstRecord.getSchema());
+				
+				try (final OutputStream out = session.write(scored);
+					final RecordSetWriter writer = writerFactory.createWriter(getLogger(), writeSchema, out, scored)
+				    ) {
+					writer.beginRecordSet();
+					
+					writer.write(scoredFirstRecord);
+					
+					Record record;
+					record = reader.nextRecord();
+					
+					while(record != null) {
+						logger.info("processing next record in the stream");
+						final Record scoredRecord = predict(record, model, getLogger());
+						logger.info("writing scored record");
+						writer.write(scoredRecord);
+						record = reader.nextRecord();
+					}
+					
+					writeResult = writer.finishRecordSet();
+
+					try {
+						writer.close();
+					} catch (final IOException ioe) {
+						getLogger().warn("Failed to close Writer for {}", new Object[]{scored});
+					}
+					
+					attributes.put("record.count", String.valueOf(writeResult.getRecordCount()));
+					attributes.put(CoreAttributes.MIME_TYPE.key(), writer.getMimeType());
+					attributes.putAll(writeResult.getAttributes());
+				}
+				
+				
+				scored = session.putAllAttributes(scored, attributes);
+				session.getProvenanceReporter().modifyContent(scored, "Modified With " + mojoPipelineUUID, stopWatch.getElapsed(TimeUnit.MILLISECONDS));
+				logger.debug("Scored {}", new Object[] {original});
+				
+			}
+		} catch (final Exception ex) {
+			logger.error("Unable to score {} due to", new Object[]{original});
+			session.transfer(original, REL_FAILURE);
+			if (scored != null) {
+				session.remove(scored);
+			}
+			return;
+		}
+		if (scored != null) {
+			logger.info("Transferring flow file on rel_success with bytes = " + scored.getSize());
+			session.transfer(scored, REL_SUCCESS);
+		}
+		session.transfer(original, REL_ORIGINAL);
+	}
+	
+	@SuppressWarnings("unchecked")
+	private Record predict(final Record record, MojoPipeline model, final ComponentLog logger) {
+		Map<String, Object> recordMap = (Map<String, Object>) DataTypeUtils.convertRecordFieldtoObject(record, RecordFieldType.RECORD.getRecordDataType(record.getSchema()));
+		
+		// Get an instance of a MojoFrameBuilder that will be used to make an input frame
+		MojoFrameBuilder frameBuilder = model.getInputFrameBuilder();
+		// Get an instance of a MojoRowBuilder that will be used to construct a row for this builder
+		MojoRowBuilder rowBuilder = frameBuilder.getMojoRowBuilder();
+		
+		logger.info("Processing input recordMap into input MojoFrame iframe");
+		for(Map.Entry<String, Object> recordEntry: recordMap.entrySet()) {
+			logger.info("Key = " + recordEntry.getKey() + ", Value = " + recordEntry.getValue());
```

Comment:
Not sure about this - this could lead to sensitive data being logged. Could it be debug instead?

Reply:
Yeah it could be logger.debug instead. I was using logger.info() to see what was happening with the data.

Reply:
Def should be guarded for debug being on and at debug level  if logged at all.